spec/services/checksum_validator_spec.rb

```diff
@@ -1,7 +1,126 @@
+  let(:endpoint) { Endpoint.find_by(storage_location: storage_dir) }
+  let(:object_dir) { "#{storage_dir}/#{DruidTools::Druid.new(druid).tree.join('/')}" }
+
+  context '#initialize' do
+    it 'sets attributes' do
+      druid = 'bj102hs9687'
+      cv = described_class.new(druid, storage_dir)
+      expect(cv.druid).to eq "druid:#{druid}"
+      expect(cv.endpoint).to eq endpoint
+      expect(cv.handler_results).to be_an_instance_of AuditResults
+    end
+  end
+
+  context '#validate_manifest_inventories' do
+    let(:druid) { 'bj102hs9687' }
+    let(:cv) { described_class.new(druid, storage_dir) }
+
+    it 'instantiates storage_object from druid and druid_path' do
+      expect(Moab::StorageObject).to receive(:new).with(cv.druid, a_string_matching(object_dir)).and_call_original
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls validate_manifest_inventory for each storage_object_version' do
+      sov1 = instance_double(Moab::StorageObjectVersion)
+      sov2 = instance_double(Moab::StorageObjectVersion)
+      sov3 = instance_double(Moab::StorageObjectVersion)
+      version_list = [sov1, sov2, sov3]
+      storage_object = instance_double(Moab::StorageObject, version_list: [sov1, sov2, sov3])
+      allow(cv).to receive(:storage_object).and_return(storage_object)
+      version_list.each do |storage_object_version|
+        expect(cv).to receive(:validate_manifest_inventory).with(storage_object_version)
+      end
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls AuditResults.report_results' do
+      results = instance_double(AuditResults, add_result: nil, :actual_version= => nil, :check_name= => nil)
+      allow(AuditResults).to receive(:new).and_return(results)
+      expect(results).to receive(:report_results)
+      cv.validate_manifest_inventories
+    end
+
+    context 'modify file signature in manifestInventory.xml' do
+      it 'adds a MOAB_FILE_CHECKSUM_MISMATCH result' do
+        druid = 'jj925bx9565'
+        object_dir = "#{storage_dir}/#{DruidTools::Druid.new(druid).tree.join('/')}"
+        file_path1 = "#{object_dir}/v0001/manifests/versionAdditions.xml"
+        file_path2 = "#{object_dir}/v0002/manifests/versionInventory.xml"
+        results = instance_double(AuditResults, report_results: nil, :check_name= => nil)
+        allow(AuditResults).to receive(:new).and_return(results)
+        cv = described_class.new(druid, storage_dir)
+        expect(results).to receive(:add_result).with(
+          AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, file_path: a_string_matching(file_path1), version: "v1"
+        )
+        expect(results).to receive(:add_result).with(
+          AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, file_path: a_string_matching(file_path2), version: "v2"
+        )
+        cv.validate_manifest_inventories
+      end
+    end
+
+    context 'no file element for versionInventory.xml in manifestInventory.xml' do
```

Comment:
so this context becomes:  "file missing from manifestInventory.xml", yes?   Do we need to check this for both metadata and content?  Or is one such check (above) sufficient?

Reply:
talked F2F about checks for both metadata and content